cpp/oneapi/dal/backend/interop/error_convertor.hpp

```diff
@@ -0,0 +1,473 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+
+#include "oneapi/dal/exception.hpp"
+#include "daal/include/services/error_handling.h"
+#include "daal/include/services/internal/status_to_error_id.h"
+
+namespace oneapi::dal::backend::interop {
+
+template <class local_convertor>
+void status_to_exception(const daal::services::Status &s, local_convertor alg_convertor) {
+    alg_convertor(s);
+    status_to_exception_default(s);
+}
+
+void status_to_exception_default(const daal::services::Status &s) {
+    using namespace daal::services;
+    using namespace daal::services::internal;
+
+    const ErrorID error = get_error_id(s);
+    const char* description = s.getDescription();
+
+    switch (error) {
+        case ErrorID::ErrorInconsistentNumberOfRows:
```

Comment:
Maybe this can be grouped? A lot of duplicate code.

Reply:
Yea, it's better to group them by an exception type. Most simplest one:
```cpp
switch(get_exception_id(error)) {
    case ErrorID::ErrorInconsistentNumberOfRows:
    case ErrorID::ErrorModelNotFullInitialized:
    case ErrorID::ErrorInconsistentNumberOfColumns:
    case ErrorID::ErrorCloneMethodFailed:
    ...
    case ErrorID::ErrorDFBootstrapVarImportanceIncompatible:
        throw invalid_argument(description);
    case ErrorID::ErrorIncorrectNumberOfFeatures:
    case ErrorID::ErrorIncorrectNumberOfObservations:
    ...
    case ErrorID::ErrorIncorrectNumberOfNodes:
            throw domain_error(description);
   ...
}
```